test/jdk/javax/swing/JToolBar/bug4368050.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2000, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4368050
+ * @summary Default toolbar layout manager should be serializable.
+ * @run main bug4368050
+ */
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import javax.swing.JToolBar;
+
+public class bug4368050 {
+    public static void main(String[] args) throws Exception {
+        JToolBar toolBar = new JToolBar();
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+             ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(toolBar);
+            byte[] buf = baos.toByteArray();
+            try (ByteArrayInputStream bais = new ByteArrayInputStream(buf);
+                 ObjectInputStream ois = new ObjectInputStream(bais)) {
+                ois.readObject();
+            }
```

Comment:
Would be better to close the stream objects in finally block in case of Exception.

```
finally  {
     if (oos != null) {
         oos.close();
     }
     if (ois != null) {
          ois.close();
     }
  } 
```

Reply:
> Better to close the stream objects in finally block in case of Exception.
> 
> ```
> finally  {
>      if (oos != null) {
>          oos.close();
>      }
>      if (ois != null) {
>           ois.close();
>      }
>   } 
> ```

No, *it's not better.* If `oos.close()` throws an exception, then `ois` is left open. In addition to that, any exception thrown from the finally block will replace the exception thrown from the try block.

*The best way is using [**try-with-resources**](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)*: it handles all the above correctly.

Reply:
Right, I looked into Aleksey's suggestion for try-with-resources and learned that this should handle things correctly. This is applicable when objects like streams or writers need to be closed for example. Thanks for taking a look!